adsft/reader.py

```diff
@@ -32,11 +33,15 @@ def read_file(input_filename, json_format=True):
     :return: File content
     """
 
-    with open(input_filename, 'r') as input_file:
-        if json_format:
-            content = json.load(input_file)
-        else:
+    if input_filename.endswith('gz'):
+        with gzip.open(input_filename, 'rb') as input_file:
             content = input_file.read().decode('utf-8')
+    else:
+        with open(input_filename, 'r') as input_file:
+            if json_format:
+                content = json.load(input_file)
+            else:
+                content = input_file.read().decode('utf-8')
```

Comment:
We have an asymmetry here: A developer could call this read function with a filename that may or may not end in `.gz` and the developer could pass `json_format` set to `True` or `False`, but `json_format` set to `True` will only work as expected if the filename is not compressed.

I understand we are not compressing the `meta.json` files now and this is not necessary at this point, but to avoid future bugs (or developers asking themselves why the json content was not parsed), I suggest we implement also the `json.load` in the compress branch of this if block.